### ISSUE TITLE
Unable to open file \tmp\index.js

### ISSUE THREAD

**Issue by mehulmpt** [CodinGame/monaco-vscode-api#164]
Thank you for the fantastic work on this library. I'm using Next.js to bundle this (which is using webpack internally) and this full setup works fine - except on Windows. On windows, here is how it crashes inside edge browser:

![Screenshot 2023-08-16 150256](https://github.com/CodinGame/monaco-vscode-api/assets/10388889/8d26a974-1710-4c83-8495-eec2ba8918e1)

This is the minimal reproduction:
1. Clone this repo: https://github.com/mehulmpt/next-monaco-demo
2. Do `yarn install`
3. Open on edge on windows

Or visit this URL where it is already deployed, on edge on windows: https://nextdemo-5yofq3tfg-mehulmpt.vercel.app/

What is wrong here exactly - I cannot figure out. Could you help?

**Comment by CGNonofr** [comment]
Indeed, it was not tested on windows

Is there the same error [on the demo here](https://codingame.github.io/monaco-vscode-api/)?

I'll guess I'll need to find a window somewhere

**Comment by mehulmpt** [comment]
The official demo works fine, but the above build fails. I couldn't understand if the bundler could cause this because my demo above works fine when used on macOS, Linux (chrome, safari, Firefox)

**Comment by kaisalmen** [comment]
@mehulmpt I can confirm your observation with Chrome / Edge on Windows, but it only occurs with your stack. We do similar things even with react in [monaco-languageclient](https://github.com/TypeFox/monaco-languageclient) and no issues there.

**Comment by mehulmpt** [comment]
Yes, it's working fine with the official demo setup. I wonder why because it crashes on the in-memory filesystem and only on specific conditions (doesn't happen on chrome on mac for example) which is weird.

**Comment by mehulmpt** [comment]
I gave it another shot today morning and figured it out. Next.js/webpack injects `process` variable inside browser itself which causes confusion in this script: `node_modules/monaco-editor/esm/vs/base/common/process.js`

This script is responsible for initializing the vscode process as win32 or posix. However, because `typeof process` is not really undefined (when it should have been - like it is in your demo example), it thinks it is running inside native Node.js environment.

After this, we're probably firing off code branches depending on if you're reading from process or from variables like `isWindows` which use user agent to determine which causes those crashes.

TL;DR - this can also be reproduced by setting your user agent to the following windows useragent on your Mac using Next.js build because Next internally also fills `process` variable:

```
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36
```

I am not sure what the official solution should be like, but for now I have patched the file itself to force it into using browser safeProcess always. Here's the diff file if someone else gets stuck here:

```
diff --git a/node_modules/monaco-editor/esm/vs/base/common/process.js b/node_modules/monaco-editor/esm/vs/base/common/process.js
index 47b3de1..865041d 100644
--- a/node_modules/monaco-editor/esm/vs/base/common/process.js
+++ b/node_modules/monaco-editor/esm/vs/base/common/process.js
@@ -3,37 +3,14 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 import { globals, isMacintosh, isWindows } from './platform.js';
-let safeProcess;
-// Native sandbox environment
-if (typeof globals.vscode !== 'undefined' && typeof globals.vscode.process !== 'undefined') {
-    const sandboxProcess = globals.vscode.process;
-    safeProcess = {
-        get platform() { return sandboxProcess.platform; },
-        get arch() { return sandboxProcess.arch; },
-        get env() { return sandboxProcess.env; },
-        cwd() { return sandboxProcess.cwd(); }
-    };
-}
-// Native node.js environment
-else if (typeof process !== 'undefined') {
-    safeProcess = {
-        get platform() { return process.platform; },
-        get arch() { return process.arch; },
-        get env() { return process.env; },
-        cwd() { return process.env['VSCODE_CWD'] || process.cwd(); }
-    };
-}
-// Web environment
-else {
-    safeProcess = {
-        // Supported
-        get platform() { return isWindows ? 'win32' : isMacintosh ? 'darwin' : 'linux'; },
-        get arch() { return undefined; /* arch is undefined in web */ },
-        // Unsupported
-        get env() { return {}; },
-        cwd() { return '/'; }
-    };
-}
+const safeProcess = {
+    // Supported
+    get platform() { return isWindows ? 'win32' : isMacintosh ? 'darwin' : 'linux'; },
+    get arch() { return undefined; /* arch is undefined in web */ },
+    // Unsupported
+    get env() { return {}; },
+    cwd() { return '/'; }
+};
 /**
  * Provides safe access to the `cwd` property in node.js, sandboxed or web
  * environments.
  * 
 ```

**Comment by CGNonofr** [comment]
Can't you prevent `process` from being injected? starting from the v5, webpack stopped injecting "builtins" like `process`, I guess next is configured to continue doing it

**Comment by mehulmpt** [comment]
Could try, but Next.js uses it for a variety of things like injecting public envs into browser.

**Comment by killpack** [comment]
@mehulmpt Did you end up finding a fix or workaround for this? I am running into a similar issue. Your reproducing example has been deleted and I'm at a loss for how to proceed.
